sku.0/sys.server/compiled/game/script/library/beast_lib.java

```diff
@@ -1938,6 +1940,41 @@ else if (happinessAdjustment < 0.5f)
         setBeastExperience(beast, experience);
         setBCDBeastExperience(bcd, experience);
     }
+    public static void incrementBeastExperienceMission(obj_id beast, int xpAmountToGrant) throws InterruptedException {

```

Comment:
I would just add an additional argument (e.g. `boolean isMissionXP`) to the existing `incrementBeastExperience` method to handle the additional calculation given the only difference between that method and this additional method is that single line of math and we're otherwise duplicating all this other stuff.